src/cross_toml.rs

```diff
@@ -67,6 +90,52 @@ impl CrossToml {
         Ok((cfg, unused))
     }
 
+    /// Merges another [`CrossToml`] into `self` and returns a new merged one
+    ///
+    /// # Merging of `targets`
+    /// The `targets` entries are merged based on the [`Target`] keys.
+    /// If a [`Target`] key is present in both configs, the [`CrossTargetConfig`]
+    /// in `other` overwrites the one in `self`.
+    ///
+    /// # Merging of `build`
+    /// The `build` fields ([`CrossBuildConfig`]) are merged based on their sub-fields.
+    /// A field in the [`CrossBuildConfig`] will only overwrite another if it contains
+    /// a value, i.e. it is not `None`.
+    pub fn merge(&self, other: &CrossToml) -> Result<CrossToml> {
+        type ValueMap = serde_json::Map<String, serde_json::Value>;
+
+        fn to_map<S: Serialize>(s: S) -> Result<ValueMap> {
+            if let Some(obj) = serde_json::to_value(s)?.as_object() {
+                Ok(obj.to_owned())
+            } else {
+                panic!("Failed to serialize CrossToml as object");
+            }
+        }
+
+        fn from_map<D: DeserializeOwned>(map: ValueMap) -> Result<D> {
+            let value = serde_json::to_value(map)?;
+            Ok(serde_json::from_value(value)?)
+        }
+
+        // Merges both target maps
+        let mut self_targets_map = to_map(&self.targets)?;
+        let other_targets_map = to_map(&other.targets)?;
+        self_targets_map.extend(other_targets_map);
+        let merged_targets = from_map(self_targets_map)?;
+
+        // Merges build configs
+        let mut self_build_cfg_map = to_map(&self.build)?;
+        let mut other_build_cfg_map = to_map(&other.build)?;
+        other_build_cfg_map.retain(|_, v| !v.is_null());
+        self_build_cfg_map.extend(other_build_cfg_map);
+        let merged_build_cfg = from_map(self_build_cfg_map)?;
```

Comment:
can't this be unified?

they are both maps, and I assume maps can be placed in maps

Reply:
It might have to be done though now that I think about it, I assume that extending without being this specific will override every value inside `target`, even if that's not the intention


Instead, you'd probably have to descend into the map and merge. See https://docs.rs/serde-toml-merge/0.3.0/src/serde_toml_merge/lib.rs.html#58-63 for example